src/concurrency/cpu_affinity.rs

```diff
@@ -13,14 +13,15 @@ pub const MAX_CPUS: usize = 1024;
 // the actual representation depends on the target's endianness and pointer width.
 // See CpuAffinityMask::set for details
 #[derive(Clone)]
-pub(crate) struct CpuAffinityMask([u8; MAX_CPUS / 8]);
+pub(crate) struct CpuAffinityMask([u8; Self::CPU_MASK_BYTES]);
 
 impl CpuAffinityMask {
+    const CPU_MASK_BYTES: usize = MAX_CPUS / 8;
     // code depends on the exact size of this type
-    const _SIZE_ASSERT: () = assert!(std::mem::size_of::<Self>() == 128);
+    const _SIZE_ASSERT: () = assert!(std::mem::size_of::<Self>() == Self::CPU_MASK_BYTES);
```

Comment:
I don't get the point of this assertion, it is literally defined as an array of CPU_MASK_BYTES bytes... and we're not transmuting `CpuAffinityMask` around so its underlying representation does not even matter.

Reply:
we're not transmuting but the logic does use `size_of::<CpuAffinityMask>` so changing its size (by adding extra fields) would break logic in perhaps non-obvious ways. That is why this asserted agains the value 128 initially: that is just the number of bytes that `cpu_size_t` is supposed to be according to libc.

I think the better approach now is to expose the `CPU_MASK_BYTES`, and use that in the logic instead of the size of the type.

Reply:
> I think the better approach now is to expose the CPU_MASK_BYTES, and use that in the logic instead of the size of the type.

:+1: